src/builder.rs

```diff
@@ -146,6 +148,30 @@ impl<'a> TransactionBuilder<'a> {
 
         self.sign_keys.iter().for_each(|k| tx.sign_inputs(k));
 
+        tx.precompute_metadata();
+
         tx
     }
+
+    #[cfg(feature = "std")]
+    pub fn finalize_checked(
+        &mut self,
+        height: Word,
+        params: &ConsensusParameters,
+    ) -> CheckedTransaction {
+        self.finalize()
+            .check(height, params)
+            .expect("failed to check tx")
+    }
+
+    #[cfg(feature = "std")]
+    pub fn finalize_checked_without_signature(
+        &mut self,
+        height: Word,
+        params: &ConsensusParameters,
+    ) -> CheckedTransaction {
+        self.finalize()
```

Comment:
as an optimization, would it make sense to skip signature generation here? Sometimes it's useful to be able to build lots random txs quickly, and we don't care about the signatures, but the sigs take up lots of time.

Reply:
That makes sense

https://github.com/FuelLabs/fuel-tx/pull/162/commits/887f5f5f3b83c5e0cda9785233810697b138a525